docs/content/config/best-practices/dkim_dmarc_spf.md

```diff
@@ -0,0 +1,341 @@
+
+DKIM requires a public/private key pair to enable **signing (_via private key_)** your outgoing mail, while the receiving end must query DNS to **verify (_via public key_)** the signature is trustworthy.
+
+### Generating Keys
+
+You should have:
+
+- At least one [email account setup][docs-accounts-add].
+- Attached a [volume for config][docs-volumes-config] to persist the generated files to local storage.
+
+DKIM is currently supported by either OpenDKIM or Rspamd:
+
+=== "OpenDKIM"
+
+    OpenDKIM is currently [enabled by default][docs-env-opendkim].
+
+    The command `docker exec <CONTAINER NAME> setup config dkim help` details supported config options, along with some examples.
+
+    !!! example "Create a DKIM key"
+
+        Generate the DKIM files with:
+
+        ```sh
+        docker exec -ti <CONTAINER NAME> setup config dkim
+        ```
+
+        Your new DKIM key(s) and OpenDKIM config files have been added to `/tmp/docker-mailserver/opendkim/`.
+
+    !!! note "LDAP accounts need to specify domains explicitly"
+
+        The command is unable to infer the domains from LDAP user accounts, you must specify them:
+
+        ```sh
+        setup config dkim domain 'mail.example.com,mail.example.io'
+        ```
+
+    !!! tip "Changing the key size"
+
+        The private key presently defaults to RSA-4096. Some DNS services and clients are only compatible with a smaller size.
+
+        You can override the default like so for 2048-bit keysize:
+
+        ```sh
+        setup config dkim keysize 2048
+        ```
+
+=== "Rspamd"
+
+    Opt-in via [`ENABLE_RSPAMD=1`][docs-env-rspamd] (_and disable the default OpenDKIM: `ENABLE_OPENDKIM=0`_).
+
+    Rspamd provides DKIM support through two separate modules:
+
+    1. [Verifying DKIM signatures from inbound mail][rspamd-docs-dkim-checks] is enabled by default.
+    2. [Signing outbound mail with your DKIM key][rspamd-docs-dkim-signing] needs additional setup (key + dns + config).
+
+    !!! example "Create a DKIM key"
+
+        Presently only OpenDKIM is supported with `setup config dkim`. To generate your DKIM key and DNS files you'll need to specify:
+
+        - `-s` The DKIM selector (_eg: `mail`, it can be anything you like_)
+        - `-d` The domain-part of email addresses you'll be sending from (_eg: `user@example.com` => `example.com`, not the mail server hostname `mail.example.com`_).
+
+        See `rspamadm dkim_keygen -h` for an overview of the command options.
+
+        1. Go inside the container with `docker exec -ti <CONTAINER NAME> bash`.
+        2. Add `rspamd/dkim/` folder to your config volume and switch to it: `cd /tmp/docker-mailserver/rspamd/dkim`
+        3. Run: `rspamadm dkim_keygen -s mail -b 2048 -d example.com -k mail.private > mail.txt` (_change `-d` to your domain-part_).
+        4. Presently you must ensure Rspamd can read the `<selector>.private` file, run:
+            -`chgrp _rspamd mail.private`
+            -`chmod g+r mail.private`
```

Comment:
```suggestion
    !!! example "Create a DKIM key"

        Presently only OpenDKIM is supported with `setup config dkim`. To generate your DKIM key and DNS files you'll need to specify:

        - `-s` The DKIM selector (_eg: `mail`, it can be anything you like_)
        - `-d` The sender address domain (_everything after `@` from the email address_)

        See `rspamadm dkim_keygen -h` for an overview of the supported options.

       ---

        1. Go inside the container with `docker exec -ti <CONTAINER NAME> bash`
        2. Add `rspamd/dkim/` folder to your config volume and switch to it: `cd /tmp/docker-mailserver/rspamd/dkim`
        3. Run: `rspamadm dkim_keygen -s mail -b 2048 -d example.com -k mail.private > mail.txt` (_change `-d` to your domain-part_)
        4. Presently you must ensure Rspamd can read the `<selector>.private` file, run:
             -`chgrp _rspamd mail.private`
             -`chmod g+r mail.private`
```